redbot/cogs/downloader/downloader.py

```diff
@@ -98,14 +98,19 @@ async def red_delete_data_for_user(self, **kwargs):
 
     def create_init_task(self):
         def _done_callback(task: asyncio.Task) -> None:
-            exc = task.exception()
-            if exc is not None:
+            try:
+                exc = task.exception()
+            except asyncio.CancelledError:
+                pass
+            else:
+                if exc is None:
+                    return
                 log.error(
                     "An unexpected error occurred during Downloader's initialization.",
                     exc_info=exc,
                 )
-                self._ready_raised = True
-                self._ready.set()
+            self._ready_raised = True
+            self._ready.set()
```

Comment:
Wouldn't this make it so that some methods don't properly wait until this task is complete?

Reply:
`initialize()` calls `self._ready.set()`

Reply:
Actually no, looking closer this would set the event before initialize is actually done. So anything waiting on `_ready` won't properly wait since this is now outside the done callback. The same exists on Mutes now too.

Reply:
Oh no I am blind on the gh diff.